tests/update_workflow.go

```diff
@@ -5204,3 +5217,151 @@ func (s *FunctionalSuite) TestUpdateWorkflow_UpdateMessageInLastWFT() {
 	6 WorkflowExecutionCompleted
 	`, s.getHistory(s.namespace, tv.WorkflowExecution()))
 }
+
+func (s *FunctionalSuite) TestUpdateWorkflow_NewSpeculativeWorkflowTask_QueryFailureClearsWFContext() {
```

Comment:
Nice, it's really helpful to have the PR come with a concrete example of one of the things it's fixing.